### DISCUSSION TITLE
Testing e-mail without access to the backend

### DISCUSSION THREAD

**Discussion by APNeto** [cypress-io/cypress#17211] [Questions and Help]
Hello, there.
I am currently testing an web app and I would like to check upon an e-mail sent from the application to the end user, though without being able to change the backend code.
Manually I am sure the app is sending the mail to destination I specify, but I am using a Gmail account for that. Is there a way I could automate this test?

Thank you.

**Comment by portyu9** [comment]
Hi @APNeto 

Yes. You do not need access to the application's backend if you can control the destination email address.

I would not automate the Gmail web UI. Instead, send the email to a dedicated test mailbox and query that mailbox programmatically.

If you want to keep using Gmail, use the Gmail API from Cypress's Node side, for example through `cy.task()`:

```js
// cypress.config.js
const { defineConfig } = require('cypress')

module.exports = defineConfig({
  e2e: {
    setupNodeEvents(on, config) {
      on('task', {
        async findEmail({ to, subject }) {
          // Query Gmail API here using a test account.
          // Return the matching message, or null if it has not arrived yet.
        },
      })
    },
  },
})
```

Then in the test:

```js
// perform the UI action that sends the email

cy.task('findEmail', {
  to: 'your-test-account@gmail.com',
  subject: 'Expected subject',
}).then((email) => {
  expect(email).to.exist
})
```

The Gmail API supports searching messages with the same type of query syntax used by Gmail, for example by sender, recipient, subject, or unread status.

Another good option is a dedicated test-inbox service such as Mailosaur/MailSlurp, which exposes an API specifically for automated email testing.

One important distinction: `cy.intercept()` can prove that the frontend made the request that should trigger the email, but it cannot prove that the email was actually delivered. To verify delivery, you need access to the receiving mailbox or an email-service API.

So in your situation, a dedicated test mailbox + API access is the cleanest solution and does not require changing the application's backend.
